src/coord/rmr/rmr.c

```diff
@@ -508,9 +508,12 @@ void MRIteratorCallback_ResetTimedOut(MRIteratorCtx *ctx) {
 
 void MRIteratorCallback_Done(MRIteratorCallbackCtx *ctx, int error) {
   // Mark the command of the context as depleted (so we won't send another command to the shard)
+  RS_ASSERT(!ctx->cmd.depleted);
   ctx->cmd.depleted = true;
   short pending = --ctx->it->ctx.pending; // Decrease `pending` before decreasing `inProcess`
-  RS_LOG_ASSERT(pending >= 0, "Pending should not reach a negative value");
+
+  RS_LOG_ASSERT_FMT(pending >= 0, "Pending (%d) should not reach a negative value. inproc: %d, freeflag: %d, channel size: %d, shard_slot: %d",
+  pending, ctx->it->ctx.inProcess, ctx->it->ctx.freeFlag, MRChannel_Size(ctx->it->ctx.chan), ctx->cmd.targetSlot);
```

Comment:
Can we always print it in debug mode rather than only when the assertion fails? perhaps it can bring more information when time comes.